src/main/java/org/broadinstitute/hellbender/utils/test/VariantContextTestUtils.java

```diff
@@ -70,10 +79,139 @@ static Object normalizeScientificNotation(final Object attribute){
+        for (int i = 0; i < newGT.size(); i++) {
+            newGT.replace(new GenotypeBuilder(newGT.get(i)).GQ(vc.getGenotype(i).getGQ()).phased(vc.getGenotype(i).isPhased()).alleles(vc.getGenotype(i).getAlleles()).make());
+        }
+
+        result.attributes(newAttributes).genotypes(newGT);
+
+        return result.make();
+
+    }
+
+    private static Object updateAttribute(final String key, final Object value,
+                                          final List<Allele> originalAlleles, final List<Allele> sortedAlleles,
+                                          final VCFHeaderLineCount count, int ploidy) {
+        if (key.startsWith("AS_")) {
+            return remapASValues((String) value, createAlleleIndexMap(originalAlleles, sortedAlleles));
+        }else {
+            switch (count) {
+                case INTEGER:
+                    return value;
+                case UNBOUNDED:
+                    //doesn't depend on allele ordering
+                    return value;
+                case A:
+                    return remapATypeValues(attributeToList(value), createAlleleIndexMap(originalAlleles, sortedAlleles));
+                case R:
+                    return remapRTypeValues(attributeToList(value), createAlleleIndexMap(originalAlleles, sortedAlleles));
+                case G:
+                    return remapGTypeValues(attributeToList(value), originalAlleles, ploidy, createAlleleIndexMap(originalAlleles, sortedAlleles));
+                default:
+                    throw new GATKException("found unexpected vcf header count type: " + count);
+            }
+        }
+    }
+
+    static List<Integer> createAlleleIndexMap(final List<Allele> originalAlleles, final List<Allele> sortedAlleles){
+        final List<Integer> mapping = new ArrayList<>(originalAlleles.size());
+        for ( final Allele a: originalAlleles){
+            final int newIndex = sortedAlleles.indexOf(a);
+            mapping.add(newIndex);
+        }
+        return mapping;
+    }
+
+    static List<Object> remapRTypeValues(List<?> oldValue, List<Integer> mapping){
+        return remapListValues(oldValue, mapping, 0);
+    }
+
+    private static List<Object> remapListValues(List<?> oldValue, List<Integer> mapping, int offset) {
+        Utils.validate(oldValue.size() + offset == mapping.size(), "attribute list size doesn't match mapping length - offset");
+        final ArrayList<Object> reordered = new ArrayList<>(oldValue.size());
+        for(int i = 0; i < oldValue.size(); i++){
+            reordered.add(oldValue.get(mapping.get(i+offset) - offset));
+        }
+        return reordered;
+    }
+
+    static List<Object> remapATypeValues(List<?> oldValue, List<Integer> mapping){
+        return remapListValues(oldValue, mapping, 1 );
+    }
+
+    static List<Object> remapGTypeValues(List<?> oldValue, List<Allele> originalAlleles, int ploidy, List<Integer> alleleIndexMap){
```

Comment:
lets just delete this if we don't trust it and it doesn't do anything

Reply:
I fixed it and added testing for the GTypeValues